File: 01-Sorting-Basic/Code-Python/insertion_sort_advance/selection_sort.py
"""
排序算法（由小到大）
"""


def sort(arr):
    """由小到大排序"""
    for i in range(len(arr) - 1):
        min_index = i
        for j in range(i + 1, len(arr)):
            if arr[min_index] > arr[j]:
                min_index = j
        if min_index != i:
            arr[min_index], arr[i] = arr[i], arr[min_index]

File: 01-Sorting-Basic/Code-Java/src/sortingBasic/insertionSortAdvance/SelectionSort.java
package sortingBasic.insertionSortAdvance;

/**
 * 选择排序（升序）
 */
public class SelectionSort {
    public static void newSort(Comparable[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int minIndex = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j].compareTo(arr[minIndex]) < 0)
                    minIndex = j;
            }
            swap(arr, i, minIndex);
        }
    }

    //交换数组中两个数
    public static void swap(Object[] arr, int index1, int index2) {
        Object temp = arr[index1];
        arr[index1] = arr[index2];
        arr[index2] = temp;
    }

}